akka-docs/rst/scala/http/client-side/websocket-support.rst

```diff
@@ -3,6 +3,61 @@
 Client-Side WebSocket Support
 =============================
 
-Not yet implemented see 17275_.
+Client side WebSocket support is available through ``Http().singleWebSocketRequest()`` ,
+``Http().webSocketClientFlow()`` and ``webSocketClientLayer``. As a WebSocket consists of two flows of
+messages of which either might happen first, rather than the request and response of a regular HTTP request,
+WebSocket requests always works with ``Flow[Message, Message, Mat]` where the messages will pass through.
+
+A WebSocket request starts with a regular HTTP request which contains an ``Upgrade`` header, so in
+addition to the flow of messages there also is an initial response from the server, this is modelled
+with ``WebSocketUpgradeResponse``.
+
+
+Message
+-------
+Messages sent and received over a WebSocket can be either ``TextMessage``s or ``BinaryMessage``s and each
+of those has two subtypes ``Strict`` or ``Streaming``. Most commonly messages will probably be ``Strict``.
```

Comment:
most commonly + probably sounds weird ;)
In typical applications messages will mostly be Strict, as WebSockets are usually deployed to communicate using small messages and not stream data, that however is possible....